progress-bar.js

```diff
@@ -0,0 +1,93 @@
+(function(){
```

Comment:
We wrap the file in an anonymous function to keep all the variables we define with `var` from 'leaking' into the global scope of the page.